### DISCUSSION TITLE
Bundle to GitHub repo mapping

### DISCUSSION THREAD

**Discussion by elsazac** [eclipse-platform/eclipse.platform.ui#1587] [Q&A] [answered]
Many times I face situations wherein, I need to know the source location of the bundle I am debugging.its trivial for familiar bundles but sometimes I have to go around different repos and test my luck.Is there any known solution for this? 
Is it worthwhile to store the GitHub information in the bundle manifest as a comment.? (not sure if manifest allows comments in the first place). Thanks in advance.

**Comment by merks** [comment]
Oomph indexes all the Eclipse GitHub repositories such that Navigate -> Open Discovered Type... can be used to find the repository of any class.  Of course if you use the general SDK setup you'll have all the sources and their clones in the IDE already:

https://github.com/eclipse-platform/.github/blob/main/CONTRIBUTING.md#creating-an-eclipse-development-environment

**Comment by akurtakov** [comment, marked as answer]
@elsazac This information is already in the Manifest . Look for the Eclipse-SourceReferences key. E.g. org.eclipse.core.resources_3.20.100.v20240122-1111.jar has:
```
Eclipse-SourceReferences: scm:git:https://github.com/eclipse-platform/ec
 lipse.platform.git;path="resources/bundles/org.eclipse.core.resources";
 tag="I20240122-1800";commitId=d60aab8335192ab8bfe01cbb1108ba4253f3023a
```

**Reply by merks** [reply]
FYI, that is information is generated by the Tycho build when configured to do so, which the Eclipse TLP build does for all its projects.

**Comment by mickaelistria** [comment]
>  This information is already in the Manifest . Look for the Eclipse-SourceReferences key

Additionally, when the bundle have such metadata, you directly import the source of the bundle from the "Plugins" view in a few clicks, as described at the bottom of https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.pde.doc.user%2Ftasks%2Fui_import_from_cvs.htm
